net/nat/nat.c

```diff
@@ -35,6 +35,22 @@
 
 #ifdef CONFIG_NET_NAT
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define NEXT_PORT(nport, hport) \
```

Comment:
Perhaps we can use the inline function to handle the following logic together:
```
net/nat/ipv4_nat_entry.c:114:      if (hport >= CONFIG_NET_DEFAULT_MAX_PORT ||
net/nat/ipv4_nat_entry.c:206:              if (hid >= CONFIG_NET_DEFAULT_MAX_PORT ||
net/udp/udp_conn.c:544:                        (CONFIG_NET_DEFAULT_MAX_PORT -
net/udp/udp_conn.c:563:      if (g_last_udp_port > CONFIG_NET_DEFAULT_MAX_PORT ||
net/tcp/tcp_conn.c:589:                        (CONFIG_NET_DEFAULT_MAX_PORT -
net/tcp/tcp_conn.c:611:          if (g_last_tcp_port > CONFIG_NET_DEFAULT_MAX_PORT ||
```

Reply:
That's why I added a separate commit, I was trying to do so in utils.h, but found they're not related to this PR, so I just wrote a simpler version here. Or maybe I can drop the commit in this PR, and change these logic together in a seperate PR after this PR is merged.

Reply:
ok, ignore my comment let us go ahead

Reply:
Hi there, a new following-up PR for TCP & UDP: https://github.com/apache/nuttx/pull/12129